src/ocaml/utils/408/load_path.ml

```diff
@@ -0,0 +1,83 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                   Jeremie Dimino, Jane Street Europe                   *)
+(*                                                                        *)
+(*   Copyright 2018 Jane Street Group LLC                                 *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+module SMap = Misc.StringMap
+
+(* Mapping from basenames to full filenames *)
+type registry = string SMap.t ref
+
+let state = Local_store.new_bindings ()
```

Comment:
The `state` value represents a list of dynamically-scoped variables.
With `srefk` create new variables, but at no point in time you redefine these variables.
See https://github.com/ocaml/merlin/blob/master/src/kernel/mocaml.ml#L30 for instance.

The original design assumed that there might be different notion of global scopes, hence the distinction between `Btype.state` and `Env.state`. 
However I think a single `typechecker_state` might be sufficient, and so this notion of binding could be removed.

I will write a PR to simplify this.

Reply:
Thanks, I rebased and this should now be fixed.

Reply:
Actually, the last commit (which uses `Local_store.Typechecker`) breaks the tests.
I guess there's something I missunderstood.